update-dependencies/Program.cs

```diff
@@ -3,152 +3,138 @@
 
 using Microsoft.DotNet.VersionTools;
 using Microsoft.DotNet.VersionTools.Automation;
+using Microsoft.DotNet.VersionTools.BuildManifest.Model;
 using Microsoft.DotNet.VersionTools.Dependencies;
+using Microsoft.DotNet.VersionTools.Dependencies.BuildOutput;
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Net.Http;
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
+using System.Xml.Linq;
 
 namespace Dotnet.Docker.Nightly
 {
     public static class Program
     {
-        private const string CliBuildInfoName = "Cli";
-        private const string SharedFrameworkBuildInfoName = "SharedFramework";
-        private static readonly string s_repoRoot = Directory.GetCurrentDirectory();
-        private static readonly Config s_config = Config.s_Instance;
-        private static bool s_updateOnly = false;
+        private static Options Options { get; set; } = new Options();
+        private static string RepoRoot { get; set; } = Directory.GetCurrentDirectory();
+        private const string RuntimeBuildInfoName = "Runtime";
+        private const string SdkBuildInfoName = "Sdk";
 
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
             try
             {
                 Trace.Listeners.Add(new TextWriterTraceListener(Console.Out));
 
-                if (ParseArgs(args))
-                {
-                    DependencyUpdateResults updateResults = UpdateFiles();
+                Options.Parse(args);
 
-                    if (!s_updateOnly && updateResults.ChangesDetected())
+                DependencyUpdateResults updateResults = await UpdateFiles();
+                if (updateResults.ChangesDetected())
+                {
+                    if (Options.UpdateOnly)
                     {
-                        CreatePullRequest(updateResults).Wait();
+                        Trace.TraceInformation($"Changes made but no GitHub credentials specified, skipping PR creation");
+                    }
+                    else
+                    {
+                        await CreatePullRequest(updateResults);
                     }
                 }
             }
             catch (Exception e)
             {
                 Console.Error.WriteLine($"Failed to update dependencies:{Environment.NewLine}{e.ToString()}");
+                Environment.Exit(1);
```

Comment:
Clarification -  please could you clarify why `Environment.Exit(1);` is needed.

Reply:
There is more than one way this can be achieved but it is a nice way to have the console app exit gracefully with a non-zero exit code.